### ISSUE TITLE
Accessibility issues with footnotes

### ISSUE THREAD

**Issue by selfthinker** [splitbrain/dokuwiki#3464]
* [ ] The footnote tooltips should have a way of closing them without using the mouse, like pressing ESC. Otherwise people who use keyboard only and don't use a mouse, like motor impaired people, will not be able close the tooltip and therefore read the text underneath it.
That fails [1.4.13 Content on Hover or Focus](https://www.w3.org/TR/WCAG21/#content-on-hover-or-focus).
![](https://user-images.githubusercontent.com/108893/113616927-ac29c980-964d-11eb-8e1f-b33a57227d13.png)

## Other non-fail accessibility issues

These don't strictly fail WCAG but are still accessibility issues. Fixing them is recommended but optional for passing WCAG.

* [ ] It would be better if the list of footnotes at the bottom of the page had their own heading. That would make it easier for screen reader users to navigate to them and know about their position early on. And it would be better for usability as it makes it easier to understand what that section is.
![](https://user-images.githubusercontent.com/108893/113617382-29553e80-964e-11eb-880b-f1c8de094a4f.png)

* [ ] Footnote links could improve their link text to make it clearer that they are linking to a footnote. That gets more important for the links at the bottom when they link back. Otherwise the links won't make sense out of context when there are, for example, two links with the link text of "1" but which go to two very different section on the page and have a different functionality.
That could be fixed by adding more information to the `title`, like "go to footnote" and "go back to where this footnote was referenced" (ideally with not that many words).

* [ ] When code blocks are read out by screen readers, the semantics of them are not read out. That means it's not clear that they are code blocks at all. Although this uses the correct semantics, annoyingly screen readers don't honour them. (Maybe there is a setting that could do that, I don't know.)
One way to fix this is by adding some explanatory text and/or heading before the code block. That could even be visually hidden.

* [ ] The file name for the example code under ["Downloadable Code Blocks" on the syntax page](https://grimm.haz.wiki/wiki:syntax#downloadable_code_blocks) is "myexample.php". Nearly all screen readers read that as one word, so it's difficult to understand. It would be better as "MyExample.php" (camel case) or "my-example.php" (with dashes). Most screen readers read that word for word. This is only a minor issue, though.
